test/commands/codepush/deployment/history-test.ts

```diff
@@ -0,0 +1,256 @@
+import * as Sinon from "sinon";
+import CodePushDeploymentHistoryCommand from "../../../../src/commands/codepush/deployment/history";
+import { models } from "../../../../src/util/apis";
+import { CommandArgs, CommandFailedResult } from "../../../../src/util/commandline";
+import { out } from "../../../../src/util/interaction";
+import { formatDate } from "../../../../src/commands/codepush/deployment/lib/date-helper";
+
+describe("CodePush deployment history", () => {
+
+  const lineFeed = "\n";
+
+  const fakeAppOwner = "test";
+  const fakeAppName = "test";
+  const fakeAppIdentifier = `${fakeAppOwner}/${fakeAppName}`;
+  const fakeNonExistentAppIdentifier = "Non/Existent";
+  const fakeDeploymentName = "Staging";
+  const fakeNonExistentDeploymentName = "Dev";
+  const fakeEmail = "fake@email.com";
+  const fakeToken = "c1o3d3e7";
+  // tslint:disable-next-line:no-http-string
+  const fakeHost = "http://localhost:1700";
+
+  const fakeCommand = new CodePushDeploymentHistoryCommand(getDeploymentHistoryCommandArgs("FAKE/FAKE", "FAKE"));
+
+  const fakeReleases: models.CodePushRelease[] = [
+    { label: "v1", releaseMethod: "Upload", description: "First Release!" },
+    { label: "v2", releaseMethod: "Promote", originalLabel: "v6", originalDeployment: "TestDep" },
+    { label: "v3", releaseMethod: "Rollback", originalLabel: "v1", isDisabled: true },
+    { label: "v4", releaseMethod: "Upload", rollout: 0 },
+    { label: "v5", releaseMethod: "Upload", rollout: 56 },
+    { label: "v6", releaseMethod: "Upload", isDisabled: true },
+    { label: "v7", releaseMethod: "Upload" },
+    { label: "v8", releaseMethod: "Upload", rollout: 0 },
+  ];
+
+  const fakeMetrics: models.CodePushReleaseMetric[] = [
+    { label: "1.0", active: 2, downloaded: 0, failed: 0, installed: 10 },
+    { label: "v1", active: 3, downloaded: 10, failed: 2, installed: 5 },
+    { label: "v2", active: 2, downloaded: 4, failed: 0, installed: 4 },
+    { label: "v5", active: 1, downloaded: 1, failed: 0, installed: 1 },
+    { label: "v6", active: 0, downloaded: 1 },
+    { label: "v7", active: 0 },
+  ];
+
+  const fakeReleasesTotalActive = fakeMetrics.reduce((sum, releaseMetrics) => sum += releaseMetrics.active, 0);
+
+  const expectedAdditionalInfoStrings = [
+    "",
+    lineFeed + "(Promoted v6 from TestDep)",
+    lineFeed + "(Rolled back v2 to v1)",
+    "",
+    "",
+    "",
+    "",
+    ""
+  ];
+
+  const expectedMetricsStrings = [
+    "Active: 38% (3 of 8)" + lineFeed + "Installed: 5 (3 pending)" + lineFeed + "Rollbacks: 2",
+    "Active: 25% (2 of 8)" + lineFeed + "Installed: 4",
+    "No installs recorded" + lineFeed + "Disabled: Yes",
+    "No installs recorded" + lineFeed + "Rollout: 0%",
+    "Active: 13% (1 of 8)" + lineFeed + "Installed: 1" + lineFeed + "Rollout: 56%",
+    "Active: 0% (0 of 8)" + lineFeed + "Disabled: Yes",
+    "Active: 0% (0 of 8)",
+    "No installs recorded" + lineFeed + "Rollout: 0%",
+  ];
+
+  const fakeReleasesResponse = [
```

Comment:
Could you please extract test data like this to separate file? It does not looks good when we have ~120 lines of test data before actual tests.

Reply:
Yes, sure, I'll do it.